test/utils/skupper/cli/revokeaccess.go

```diff
@@ -0,0 +1,155 @@
+package cli
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/test/utils"
+	"github.com/skupperproject/skupper/test/utils/base"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+)
+
+var (
+	expectUpdatedSecrets    = []string{types.SiteCaSecret, types.SiteServerSecret, types.ClaimsServerSecret}
+	expectUpdatedComponents = []string{types.RouterComponent, types.ControllerComponentName}
+)
+
+const (
+	// timeout defines how long to wait for revoke-access to complete
+	timeout = time.Minute
+)
+
+// RevokeAccessTester allows running and validating `skupper revoke-access`.
+type RevokeAccessTester struct {
+	ExpectClaimRecordsDeleted bool
+	secretInformer            cache.SharedIndexInformer
+	podInformer               cache.SharedIndexInformer
+	claimRecordsDeleted       bool
+}
+
+func (d *RevokeAccessTester) Command(cluster *base.ClusterContext) []string {
+	args := SkupperCommonOptions(cluster)
+	args = append(args, "revoke-access")
+	return args
+}
+
+func (d *RevokeAccessTester) Run(cluster *base.ClusterContext) (stdout string, stderr string, err error) {
+
+	//
+	// Creating informers to monitor secrets and pods (before revoke-access is issued):
+	// - Removal of those labeled as 'skupper.io/type=token-claim-record'
+	// - Updates to: skupper-site-ca secret, skupper-site-server, skupper-claims-server
+	// - router and service-controller pods restarted (after deployment updated)
+	//
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	doneCh := d.initializeInformers(cluster, stopCh)
```

Comment:
Out of curiousity, why use informers rather than a synchronous check after the command has run?

Reply:
The main reason is to avoid retrieving the secrets and pods before and after issuing the revoke-access
command, as I would need to compare the resourceVersion (or some other attribute) of the secrets and pods.
Using informers I just watch what I am expecting to happen and assert everything is done.

@grs Do you have any other recommendation in this case?

Reply:
No, that's fine. The only issue with the informers is that they are asynchronous, so you need to wait for some amount of time to be sure you have received necessary events (or end the test only when you get expected events). No issue from me doing it this way though, I was just curious.

Reply:
With secrets you don't need to poll. Once the revoke-access returns, the secrets should be updated or deleted. As you say though, both approaches will work.

Reply:
Correct.